src/apps/Permissions/AppPermissions.js

```diff
@@ -1,140 +1,56 @@
-                            <TableHeader />
-                          </TableRow>
-                        }
-                      >
-                        {appPermissions.map(({ role, entity }, i) => (
-                          <Row
-                            key={i}
-                            role={role}
-                            entity={entity}
-                            proxyAddress={address}
-                            onRevoke={revokePermission}
-                          />
-                        ))}
-                      </Table>
-                    )}
-                  </Viewport>
-                )}
-              </Section>
-              <EntityPermissions
-                title="Permissions granted to this app"
-                noPermissionsLabel="No permissions granted."
-                address={address}
-                loading={loading}
-                onRevoke={revokePermission}
-              />
-            </React.Fragment>
-          )
-        }}
-      </PermissionsConsumer>
-    )
-  }
-}
+function AppPermissions({ app, loading, onBack, onManageRole }) {
+  const permissions = usePermissionsByRole()
 
-class Row extends React.Component {
-  static propTypes = {
-    entity: PropTypes.object.isRequired,
-    onRevoke: PropTypes.func.isRequired,
-    proxyAddress: EthereumAddressType.isRequired,
-    role: PropTypes.object.isRequired,
+  if (loading) {
+    return <EmptyBlock>Loading permissions…</EmptyBlock>
   }
 
-  handleRevoke = () => {
-    const { onRevoke, role, entity, proxyAddress } = this.props
-    onRevoke({
-      proxyAddress,
-      roleBytes: role.bytes,
-      entityAddress: entity.address,
-    })
-  }
-  renderEntity() {
-    const { entity } = this.props
-    if (!entity) {
-      return 'Unknown'
-    }
-    if (entity.type === 'app') {
-      return <AppInstanceLabel app={entity.app} proxyAddress={entity.address} />
-    }
+  const appPermissions = permissions.filter(
+    permission =>
+      permission.app && permission.app.proxyAddress === app.proxyAddress
+  )
 
-    return (
-      <LocalIdentityBadge
-        entity={entity.type === 'any' ? 'Any account' : entity.address}
-      />
-    )
+  if (appPermissions.length === 0) {
+    return <EmptyBlock>No permissions found.</EmptyBlock>
   }
-  render() {
-    const { role } = this.props
-    return (
-      <TableRow>
-        <FirstTableCell>
-          <Text weight="bold">{role ? role.name : 'Unknown'}</Text>
-        </FirstTableCell>
-        <TableCell>{this.renderEntity()}</TableCell>
-        <LastTableCell align="right">
-          <Button
-            mode="outline"
-            emphasis="negative"
-            compact
-            onClick={this.handleRevoke}
+
+  return (
+    <React.Fragment>
+      <Bar>
+        <BackButton onClick={onBack} />
+      </Bar>
+      <PermissionsView
+        heading={
+          <span
+            css={`
+              ${textStyle('body2')}
+              font-weight: 600;
```

Comment:
Why is this needed? Shouldn't `body2` take care of the correct `font-weight`?

Reply:
The designs had them at a different `font-weight` than the default (400).

Reply:
Then my suggestion would be to revisit the designs so that these are in sync, it seems a bit arbitrary that this `body2` text has a different `font-weight`
cc @dizzypaty

Reply:
Yes we need to add some variants I think (e.g. `body2-bold`).